scripts.js

```diff
@@ -0,0 +1,545 @@
+// DOM SELECTORS
+
+const recipeGrid = document.getElementById("recipe-grid")
+const resetFilter = document.getElementById("reset-filter")
+
+// GLOBAL VARIABLES
+const URL =
+  "https://api.spoonacular.com/recipes/random?apiKey=690ac6592da546bc9d81f64e827555ff&number=10"
+
+// Empty array to store the fetched recipes
+let recipes = []
+
+// Total number of recipes fetched
+let totalRecipesFetched =
+  parseInt(localStorage.getItem("totalRecipesFetched")) || 0
+
+// List for chosen diets
+let dietFilters = []
+
+// Global variable to keep the filtered recipes
+let filteredRecipes = [...recipes]
+
+// FUNCTIONS
+
+// Save recipes to localStorage
+const saveRecipesToLocalStorage = () => {
+  localStorage.setItem("recipes", JSON.stringify(recipes))
+}
+
+// Load recipes from localStorage
+const loadRecipesFromLocalStorage = () => {
+  const storedRecipes = localStorage.getItem("recipes")
+  console.log("Loading recipes from localStorage:", recipes)
+  return storedRecipes ? JSON.parse(storedRecipes) : []
+}
+
+const updateRecipeList = (filteredRecipes) => {
+  const recipeGrid = document.getElementById("recipe-grid")
+  // Clear the grid before adding new recipes
+  recipeGrid.innerHTML = ""
+
+  // Check if there are no recipes to display
+  if (!filteredRecipes || filteredRecipes.length === 0) {
+    recipeGrid.innerHTML = `<p>No recipes found.</p>`
+    return
+  }
+
+  let likedRecipes = JSON.parse(localStorage.getItem("likes")) || []
+
+  filteredRecipes.forEach((recipe) => {
+    const recipeCard = document.createElement("div")
+    recipeCard.classList.add("recipe-card")
+
+    // Create a list of ingredients (ADDED)
+    if (!recipe.ingredients) {
+      ingredientsHTML = `<li>Ingredients not available</li>`
+    } else {
+      ingredientsHTML = recipe.ingredients
+        .map((ingredient) => `<li>${ingredient}</li>`)
+        .join("")
+    }
+
+    // Add recipe details inside the div
+    recipeCard.innerHTML = `
+    <button class="like-button" data-id="${
+      recipe.id
+    }" onclick="toggleLike(event)">
+        <img src="/assets/heart.svg" alt="Like">
+      </button>
+    <img src="${recipe.image}" alt="${recipe.title}">
+    <h3>${recipe.title}</h3>
+    <p><b>Likes:</b> ${recipe.aggregateLikes}</p>
+    <p><b>Dish type:</b> ${recipe.dishTypes.join(", ")}</p>
+    <p><b>Cuisine:</b> ${recipe.cuisines}</p>
+    <p><b>Diet:</b> ${recipe.diets.join(", ")}</p>
+    <p><b>Time:</b> ${recipe.readyInMinutes} min </p>
+    <p><b>Servings:</b> ${recipe.servings}</p>
+    <hr>
+    <p><b>Ingredients:</b></p>
+    <ul>
+    ${ingredientsHTML}
+    </ul>
+    `
+    // If a recipe is in the likedRecipes list add the class "liked" to it
+    if (likedRecipes.includes(recipe.id)) {
+      recipeCard.querySelector(".like-button").classList.add("liked")
+    }
+
+    // Append the recipe card to the recipe grid
+    recipeGrid.appendChild(recipeCard)
+
+    document.querySelectorAll(".recipe-card").forEach((card) => {
```

Comment:
Här lägger du in en forEach i skapandet av ett recipe-card. Så för varje recipie-card du skapar i den här funktionen hämtar kodblocket på rad 92 alla element med klassen .recipe-card och lägger eventlyssnare i dessa. ForEach behövs inte här om jag läst koden rätt. Alt flytta ut detta kodblock efter den här funktionen.